src/utils/str.c

```diff
@@ -209,9 +209,9 @@ bool str_first_of(const str *string, char find, size_t *pos) {
 }
 
 bool str_last_of(const str *string, char find, size_t *pos) {
-    for(size_t i = string->len; i >= 0; i--) {
-        if(string->data[i-1] == find) {
-            *pos = i - 1;
+    for(size_t i = 0; i < string->len; i++) {
+        if(string->data[string->len - i] == find) {
```

Comment:
Is the intent that this can find the last ```\0``` character too? I kind of would not expect that this function can access ```string->data``` at the ```string->len``` position since ...

Reply:
i was thinking maybe it could be changed to
```
for(size_t i = 1; i <= string->len; i++) {
    ...
```

Reply:
string->len does not account for the \0. Allocated length is always string->len + 1 :)

Reply:
Yeah, I know, was just wondering why is it that you can use this to find the ```\0``` char, but I guess this is fine.

Reply:
Ah, right, I misread your comment. Right you are, it is missing -1 on the seek. Added now and bumped in an additional test.